lib/smart_proxy_ansible/validate_settings.rb

```diff
@@ -0,0 +1,7 @@
+module Proxy::Ansible
+  class ValidateSettings < ::Proxy::PluginValidators::Base
+    def validate!(settings)
+      raise NotExistingWorkingDirException.new("Working directory does not exist") unless settings[:working_dir].nil? || File.directory?(File.expand_path(settings[:working_dir]))
```

Comment:
The logic is a little bit off. If the working dir is `nil`, it short circuits and never checks if it actually exists. We want either `if nil or not a directory` or `unless not-nil and File.directory?`

Reply:
Looks like I misread the whole thing. The directory either has to exist or be unset, which you captured here correctly. Nevermind then